vnext/ReactUWP/Modules/DevSupportManagerUwp.cpp

```diff
@@ -35,22 +35,35 @@ using namespace facebook::react;
 
 namespace react { namespace uwp {
 
-std::future<std::string> DownloadFromAsync(const std::string& url)
+std::future<std::pair<std::string, bool>> DownloadFromAsync(const std::string& url)
 {
   winrt::Windows::Web::Http::HttpClient httpClient;
   winrt::Windows::Foundation::Uri uri(facebook::react::UnicodeConversion::Utf8ToUtf16(url));
 
   co_await winrt::resume_background();
 
-  winrt::Windows::Storage::Streams::IBuffer buffer = co_await httpClient.GetBufferAsync(uri);
-  auto reader = winrt::Windows::Storage::Streams::DataReader::FromBuffer(buffer);
+  winrt::Windows::Web::Http::HttpRequestMessage request(winrt::Windows::Web::Http::HttpMethod::Get(), uri);
+  winrt::Windows::Web::Http::HttpResponseMessage response = co_await httpClient.SendRequestAsync(request);
 
-  reader.UnicodeEncoding(winrt::Windows::Storage::Streams::UnicodeEncoding::Utf8);
-  uint32_t len = reader.UnconsumedBufferLength();
-  std::vector<uint8_t> data(len);
-  reader.ReadBytes(data);
+  if (response.IsSuccessStatusCode())
+  {
+    winrt::Windows::Storage::Streams::IInputStream stream = co_await response.Content().ReadAsInputStreamAsync();
+    auto reader = winrt::Windows::Storage::Streams::DataReader(stream);
+
+    reader.UnicodeEncoding(winrt::Windows::Storage::Streams::UnicodeEncoding::Utf8);
+    uint32_t len = reader.UnconsumedBufferLength();
+    std::vector<uint8_t> data(len);
+    reader.ReadBytes(data);
 
-  co_return std::string(reinterpret_cast<char*>(data.data()), data.size());
+    co_return std::make_pair(std::string(reinterpret_cast<char*>(data.data()), data.size()), true);
+  }
+  else
+  {
+    // only ReadAsString can read the error text out of Content()
+    winrt::hstring result = co_await response.Content().ReadAsStringAsync();
```

Comment:
When I debug it, the response is 500 error with json output.
Can you try to parse output as json first, then get the errors part? You may don't need the pretty part in this way.
Also it's usefully it we includes the status code in the error message.

Reply:
I could do that but the errors property is an array of content that is again still escaped, I'd have to bake in knowledge of that specific json payload.  I agree the http error code might be useful in the error text if we didn't have json.  we were kind of by chance getting a string that mentioned 500 before, what was happening an hresult_error came back and windows had an hresult for that specific error...  that hresult error was coming from using the simpler httpclient API.